Installation fails on current `main` branch

If you follow the instructions to install Odyssey from source using the current `main` branch:
```
git clone https://github.com/ithacaxyz/odyssey
cd odyssey
cargo install --path bin/odyssey
```
It fails with the following error:
```
error[E0603]: module `eip1559` is private
  --> /Users/zachobront/.cargo/git/checkouts/reth-36d3ea1d1152b20c/4b4f9cf/crates/optimism/payload/src/payload.rs:10:25
   |
10 | use op_alloy_consensus::eip1559::{decode_holocene_extra_data, EIP1559Par...
   |                         ^^^^^^^ private module
   |
note: the module `eip1559` is defined here
  --> /Users/zachobront/.cargo/registry/src/index.crates.io-6f17d22bba15001f/op-alloy-consensus-0.6.7/src/lib.rs:22:1
   |
22 | mod eip1559;
   | ^^^^^^^^^^^

error[E0603]: module `eip1559` is private
  --> /Users/zachobront/.cargo/git/checkouts/reth-36d3ea1d1152b20c/4b4f9cf/crates/optimism/payload/src/payload.rs:10:25
   |
10 | use op_alloy_consensus::eip1559::{decode_holocene_extra_data, EIP1559Par...
   |                         ^^^^^^^ private module
   |
help: consider importing this enum instead:
      op_alloy_consensus::EIP1559ParamError
  --> /Users/zachobront/.cargo/git/checkouts/reth-36d3ea1d1152b20c/4b4f9cf/crates/optimism/payload/src/payload.rs:10:63
   |
10 | ...code_holocene_extra_data, EIP1559ParamError};
   |                              ^^^^^^^^^^^^^^^^^
note: the module `eip1559` is defined here
  --> /Users/zachobront/.cargo/registry/src/index.crates.io-6f17d22bba15001f/op-alloy-consensus-0.6.7/src/lib.rs:22:1
   |
22 | mod eip1559;
   | ^^^^^^^^^^^

```
Everything works fine before the [latest commit](https://github.com/ithacaxyz/odyssey/commit/9c06a05e489992246af7a956d92543f10c46ee5b). It looks like bumping the version of `reth-optimism-payload-builder` pulls in the new `op-alloy-consensus`, which includes [this commit](https://github.com/alloy-rs/op-alloy/commit/cbac051d27b01fb85d1d628428a9ca8f9e9d9ad0) that re-exports the items from `eip1559` but makes the module itself private.

(Required as I'm trying to run a node with new reth features.)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -155,42 +155,43 @@ alloy-signer-local = { version = "0.6.4", features = ["mnemonic"] }
 tokio = { version = "1.21", default-features = false }
 
 # reth
-reth-chainspec = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-cli = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-cli-util = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-evm = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-rpc-eth-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-node-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-node-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-node-core = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-chainspec = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-cli = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-cli-util = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-evm = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-rpc-eth-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-node-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-node-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-node-core = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-optimism-node = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-optimism-node = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-optimism-cli = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-optimism-cli = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-optimism-rpc = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-optimism-rpc = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-optimism-forks = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-optimism-chainspec = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-payload-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-primitives = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-optimism-primitives = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-optimism-forks = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-optimism-chainspec = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-payload-builder = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-primitives = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-provider = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf", features = [
+reth-provider = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac", features = [
     "optimism",
 ] }
-reth-revm = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-storage-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-tracing = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-transaction-pool = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-trie-db = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-network = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-network-types = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
-reth-chain-state = { git = "https://github.com/paradigmxyz/reth.git", rev = "4b4f9cf" }
+reth-revm = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-storage-api = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-tracing = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-transaction-pool = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-trie-db = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-network = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-network-types = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
+reth-chain-state = { git = "https://github.com/paradigmxyz/reth.git", rev = "f211aac" }
 revm-primitives = { version = "14.0.0" }
 revm-precompile = { version = "15.0.0", features = ["secp256r1"] }
 

--- a/crates/node/Cargo.toml
+++ b/crates/node/Cargo.toml
@@ -19,6 +19,7 @@ reth-node-builder.workspace = true
 reth-optimism-node.workspace = true
 reth-optimism-forks.workspace = true
 reth-optimism-chainspec.workspace = true
+reth-optimism-primitives.workspace = true
 reth-chainspec.workspace = true
 reth-payload-builder.workspace = true
 reth-primitives.workspace = true

--- a/crates/node/src/node.rs
+++ b/crates/node/src/node.rs
@@ -21,9 +21,12 @@ use reth_node_builder::{
 use reth_optimism_chainspec::OpChainSpec;
 use reth_optimism_node::{
     args::RollupArgs,
-    node::{OpAddOns, OpConsensusBuilder, OpNetworkBuilder, OpPayloadBuilder, OpPoolBuilder},
+    node::{
+        OpAddOns, OpConsensusBuilder, OpNetworkBuilder, OpPayloadBuilder, OpPoolBuilder, OpStorage,
+    },
     OpEngineTypes, OpExecutionStrategyFactory,
 };
+use reth_optimism_primitives::OpPrimitives;
 use reth_payload_builder::PayloadBuilderHandle;
 use reth_transaction_pool::{SubPoolLimit, TransactionPool, TXPOOL_MAX_ACCOUNT_SLOTS_PER_SENDER};
 use reth_trie_db::MerklePatriciaTrie;
@@ -82,9 +85,10 @@ impl OdysseyNode {
 
 /// Configure the node types
 impl NodeTypes for OdysseyNode {
-    type Primitives = ();
+    type Primitives = OpPrimitives;
     type ChainSpec = OpChainSpec;
     type StateCommitment = MerklePatriciaTrie;
+    type Storage = OpStorage;
 }
 
 impl NodeTypesWithEngine for OdysseyNode {
@@ -93,7 +97,14 @@ impl NodeTypesWithEngine for OdysseyNode {
 
 impl<N> Node<N> for OdysseyNode
 where
-    N: FullNodeTypes<Types: NodeTypesWithEngine<Engine = OpEngineTypes, ChainSpec = OpChainSpec>>,
+    N: FullNodeTypes<
+        Types: NodeTypesWithEngine<
+            Engine = OpEngineTypes,
+            ChainSpec = OpChainSpec,
+            Primitives = OpPrimitives,
+            Storage = OpStorage,
+        >,
+    >,
 {
     type ComponentsBuilder = ComponentsBuilder<
         N,

